src/commands/extractText.ts

```diff
@@ -42,63 +42,102 @@ const m: ExtensionModule = () => {
         default_keypath = keyPrefix + default_keypath
 
       const locale = Config.sourceLanguage
-
-      // prompt for keypath
-      const keypath = await window.showInputBox({
-        prompt: i18n.t('prompt.enter_i18n_key'),
-        value: default_keypath,
-        ignoreFocusOut: true,
+      let keypath = default_keypath
+
+      const paths: Record<string, string> = {}
+
+      // filter leafs from key tree
+      CurrentFile.loader.keys.forEach((key) => {
+        const lastIndex = key.lastIndexOf('.')
+        let label = ''
+        if (lastIndex === -1)
+          label = key
+        else
+          label = key.substr(0, lastIndex)
+        paths[label] = ''
       })
 
-      if (!keypath) {
-        window.showWarningMessage(i18n.t('prompt.extraction_canceled'))
-        return
-      }
-
-      if (!keypathValidate(keypath))
-        return window.showWarningMessage(i18n.t('prompt.invalid_keypath'))
+      // create quickPickItems out of the paths
+      const quickPickItems: QuickPickItem[] = Object.keys(paths).map(path => ({ label: path }))
+
+      // create and init a QuickPick for the path
+      const pathPicker = window.createQuickPick()
+      pathPicker.placeholder = 'Select (or create) a Path'
+      pathPicker.ignoreFocusOut = true
+      pathPicker.canSelectMany = false
+      pathPicker.items = quickPickItems
+
+      pathPicker.onDidAccept(() => {
+        const selection = pathPicker.activeItems[0]
+        pathPicker.hide()
+        keypath = `${selection.label}.`
+        keyPicker.prompt = `Enter i18n key for the path: ${keypath}`
```

Comment:
It would be great if you can extract the texts into locales and make it i18n-able. Same with line 65 and 81. Thanks